crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1495,6 +1496,364 @@ impl<'w, 'a, T: Component> VacantEntry<'w, 'a, T> {
     }
 }
 
+/// Provides read-only access to a single entity and some of its components defined by the contained [`Access`].
+pub struct FilteredEntityRef<'w> {
+    entity: UnsafeEntityCell<'w>,
+    access: Access<ComponentId>,
+}
+
+impl<'w> FilteredEntityRef<'w> {
+    /// # Safety
+    /// - `cell` must have permission to mutate every component of the entity.
+    /// - No `&mut World` can exist from the underlying `UnsafeWorldCell`
+    /// - If `access` takes a read on a component, no mutable references to that component
+    /// may exist at the same time as the returned [`FilteredEntityRef`].
```

Comment:
The first invariant here doesn't look right.

Reply:
Correct, incorrectly copied from EntityRef.

Reply:
So does that mean `EntityRef`'s invariants are wrong?

Reply:
Yes, actual source was EntityMut, my bad, was to quick in my response.